numba/targets/arrayobj.py

```diff
@@ -294,45 +273,110 @@ def _getitem_array_generic(context, builder, return_type, aryty, ary,
     output_shapes = []
     output_strides = []
 
-    for ax, (indexval, idxty) in enumerate(zip(indices, index_types)):
+    ax = 0
+    for indexval, idxty in zip(indices, index_types):
+        if idxty is types.ellipsis:
+            # Fill up missing dimensions at the middle
+            n_missing = aryty.ndim - len(indices) + 1
```

Comment:
Despite the doc (http://docs.scipy.org/doc/numpy/reference/arrays.indexing.html) saying that can only be one ellipsis, numpy silently converts all remaining ellipsis to a `:` (all elements in the axis).  Only the first ellipsis is expanded.  See http://sourceforge.net/p/numpy/mailman/message/12594688/

The code here does not handle the remaining ellipsis.

The following will fail:

```python
from numba import njit
import numpy as np


arr = np.arange(200).reshape(5, 2, 2, 2, 5)

@njit
def foo(a, i, j):
    return a[i, ..., j, ...]


print(foo(arr, 1, 1))
```